src/utils/Converters.py

```diff
@@ -0,0 +1,38 @@
+from __future__ import annotations
+from discord.ext import commands
+import re
+
+
+class Course:
+    def __init__(self, *, dept: str, course: str):
+        self._dept: str = dept.upper()
+        self._course: str = course.upper()
+
+    @property
+    def dept(self) -> str:
+        return self._dept
+
+    @property
+    def course(self) -> str:
+        return self._course
+
+    @property
+    def year_level(self) -> str:
+        return self._course[0]
+
+    def __str__(self) -> str:
+        return f"{self.dept} {self.course}"
+
+    @classmethod
+    async def convert(cls, ctx: commands.Context, argument: str) -> Course:
+        if len(argument) > 8:
```

Comment:
I assume `8` comes from the maximum number of characters possible in a course? (eg. `CPEN400D`). Can we make this a constant or add a comment?

Reply:
Yeah, it came from the [original course converter](https://github.com/kelvinkoon/ecess-discord-bot/blob/main/src/cogs/PrequisiteChecker.py#L15) which was inspired by [this](https://github.com/kelvinkoon/ecess-discord-bot/blob/main/src/cogs/PrequisiteChecker.py#L62). Agreed that it's a magic value though -- I'll extract it out.

Reply:
Yeah, whoever didn't catch that in code review didn't do a very good job. Totally wasn't me :')
Appreciate moving it out.